monteCarloTreeSearch.c

```diff
@@ -0,0 +1,334 @@
+
+	startTime = clock();				//track time limit
+	initializeRoot(root, gameBoard);	  // initialize root
+	newNode = root;  					// temporary newNode
+	selected = root;					//initialize selected value for expansion
+
+	while(numNewNodes < nodeLimit && flag > 0)
+	{
+		gameContinues = 100;	//initialize the simulation phase gameloop value
+		bestValue = -1000;		// initialize bestvalue as a negative number
+		numOfActions = 0;		//set numOfActions to zero each loop through
+
+		//1. Select a node using UCT and the functions provided by #1. (See the js implementation of 
+		//selection and the js UCT implementation as a guide).
+
+		//UCT calculation to select next node 
+		for(i = 0; i < newNode->numChildren; i++)
+		{
+			child = newNode->children[i];
+			//UCT calculation
+			uctValue = child->numWins / child->numSimulations + 
+			sqrt(2 * log(newNode->numSimulations / child->numSimulations));
+
+			//check if the uct value for each child is better than the one before
+			if(uctValue > bestValue)
+			{
+				selected = child;
+				bestValue = uctValue;
+			}
+		} 
+
+
+		
+		getMoves(gameBoard, allActions);
+		//2. Expand the selected node using the expand() function (see #1) after choosing a move. 
+		//It can be any unplayed move given by getMoves() (see #2).
+		for(i = 0; i < 200; i++)
+		{
+			if(allActions[i] == -1)
+			{
+				break;
+			}
+
+			numOfActions++;
+		}
+		if(numOfActions != 0)
+		{
+
+			for(i = 0; i < numOfActions; i+2)
+			{
+			
+				action[0] = allActions[i];
+				action[1] = allActions[i+ 1];
+				//You need to implement a choose move here and then send that move as action.
+				expand(selected, action);
+				numNewNodes++;
+
+			}
+		}
+		// else
+		// {
+		// 	move[0] = -1;
+		// 	move[1] = -1;
+		// 	flag == -100;
+		//	break;
+		// }
+
+		//3. Simulate the game board on the new node until the game is complete 
+		for(j = 0; j < selected->numChildren; j++)
+		{
+			boardState = selected->board;
+			while(gameContinues > 0)
+			{
+				numOfActions = 0;
+				//get board structure from the node call getmoves and for loop with while loop inside
+				//placePiece()
+				//get the move to be made and play the game until the player wins or loses
+				getMoves(boardState, allActions);
+
+				for(i = 0; i < 200; i++)
+				{
+					if(allActions[i] == -1)
+					{
+						break;
+					}
+
+					numOfActions++;
+				}
+				//make sure that there are still actions to be taken before the game ends
+				if(numOfActions != 0)
+				{
+					//Randomly select a move to make
+					randomValue = numOfActions / 2;
+					randomValue = rand() % randomValue + 1;  //may not be plus 1
+					boardMove[0] = allActions[randomValue];
+					boardMove[1] = allActions[randomValue + 1]; 
+				}
+				else
+				{
+					boardMove[0] = -1;
+					boardMove[1] = -1;
+				}
+				//player cannot make a move
+				if(boardMove[0] == -1 && boardMove[1] == -1)
+				{
+
+					//Check if other player can move
+					flipMove(boardState->whoseMove);
```

Comment:
Note: if you really want me to give you access to the `flipMove()` function you're gonna need to pass a reference to whoseMove instead of passing by value. Let me know if you need me to expose `flipMove()`.